# 2. Exersice Sheet
von Celina Kortmann, Felix Göke und Erich Wagner

## Exersice 3 - Dice 
You roll two dice, one red and one blue. What is the probability that  
(a) the sum of pips is 9?  
Es gibt insgesamt 4 Möglichkeiten, wie die Augenzahl der beiden Würfel 9 ergeben kann. Es gibt die Kombination (3,6) und (4,5) und jeweils andersrum. Da die Wahrscheinlichkeiten unabhängig voneinander sind (der rote Würfel beeinflusst den blauen nicht und andersrum) gilt: $P(A \wedge B)=P(A) \cdot P(B)$. Außerdem verwenden wir das 3. Kolmogorov-Axiom.
$$
\begin{aligned}
P(W_r+W_b=9)&=P(W_r=3 \wedge W_b=6)+P(W_r=6 \wedge W_b=3)\\
&+P(W_r=4 \wedge W_b=5)+P(W_r=5 \wedge W_b=4) \\
&=2\cdot P(W_r=3 \wedge W_b=6) + 2\cdot P(W_r=4 \wedge W_b=5)\\
&=2 \cdot P(W_r=3)\cdot P(W_b=6)+2 \cdot P(W_r=4)\cdot P(W_b=5)\\
&=2 \cdot \frac{1}{6}\cdot \frac{1}{6}+2 \cdot \frac{1}{6}\cdot \frac{1}{6}\\
&=\frac{4}{36}=\frac{1}{9}
\end{aligned}
$$
(b) the sum of pips is 9 or larger?  
Jetzt müssen wir außerdem die Wahrscheinlichkeit dafür betrachten, dass die Augenzahl 10, 11 oder 12 ist und dann die Summe bilden. Für $W_{ges}=12$ gibt es 1 Möglichkeit, für $W_{ges}=11$ gibt es 2 Möglichkeiten. Für $W_{ges}=10$ gibt es 3 Möglichkeiten.
$$
\begin{aligned}
P(W_r+W_b \geq 9)=&P(W_r+W_b=9)+P(W_r+W_b=10)+P(W_r+W_b=11)+P(W_r+W_b=12) \\
&=\frac{4}{36}+\frac{3}{36}+\frac{2}{36}+\frac{1}{36} \\
&=\frac{10}{36}
\end{aligned}
$$
(c) one dice shows 4 and the other 5 pips?
$$
\begin{aligned}
P((W_r=4 \wedge W_b=5) \vee (W_r=5 \wedge W_b=4))&=P(W_r=4 \wedge W_b=5) + P(W_r=5 \wedge W_b=4) \\
&= P(W_r=4) \cdot P(W_b=5) + P(W_r=5) \cdot P(W_b=4) \\
&= \frac{1}{6} \cdot \frac{1}{6} + \frac{1}{6} \cdot \frac{1}{6} \\
&= \frac{2}{36} \\
&= \frac{1}{18}
\end{aligned}
$$
(d) the red dice shows 4 and the blue dice 5 pips?
$$
\begin{aligned}
P(W_r=4 \wedge W_b=5)&= P(W_r=4) \cdot P(W_b=5) \\
&=\frac{1}{6}\cdot \frac{1}{6} \\
&= \frac{1}{36}
\end{aligned}
$$


Your roll the dice so that the blue dice rolls behind an object so that you cant see it first. The red dice shows a 4. After you have seen this, what is the prabability that  
(e) the sum of pips is nine?  
Wenn wir wissen, dass der rote Würfel eine 4 zeigt, dann suchen wir nach der Wahrscheinlichkeit, dass der blaue Würfel eine 5 zeigt, damit die Gesamtaugenzahl 9 ist. Da die Wahrscheinlichkeiten unabhängig voneinander sind, folgt:
$$
P(W_b=5| W_r=4)=P(W_b=5)=\frac{1}{6}
$$

(f) the sum of pips is 9 or larger?  
$$
\begin{aligned}
P(W_b\geq 5|W_r=4)&=P(W_b \geq 5) \\
&=P(W_b=5 \vee W_b=6) \\
&=P(W_b=5)+P(W_b=6) \\
&=\frac{1}{6}+\frac{1}{6} \\
&=\frac{2}{6}\\
&=\frac{1}{3}
\end{aligned}
$$

(g) the red dice shows 4 and the blue dice 5 pips?  
This is the same as (e), weil $W_b=5$ gleichbedeutend mit einer Gesamtaugenzahl von 9 ist, mit dem Wissen, dass der rote Würfel eine 4 zeigt, was unsere Annahme für die Aufgabenteile (e), (f) und (g) ist.

## Exercise 4 - Maxwell velocity distribution 
The probability density of the magnitude of the velocity 𝑣 of the molecules in an ideal gas at the
absolute temperature 𝑇 is given by
$$
f(v)=N\cdot \exp \left( -\frac{m v^2}{2k_B T} \right) \cdot 4 \pi v^2,
$$
where 𝑚 is the molecular mass, $k_B$
is the Boltzmann constant, and 𝑁 is the normalization constant.

Zunächst muss die Teilchenzahl bestimmt werden, als Funktion von $m$ und $T$. Es gilt:
$$
\int_{0}^{\infty} f(v)dv=1,
$$
da die sichere Wahrscheinlichkeit 1 sein muss. Aus dieser Normierungsbedingung ergibt sich für N in drei Dimensionen:
$$
N=\left(\frac{m}{2 \pi k_B T}\right)^{3/2}
$$
Für die Maxwellverteilung gilt somit
$$
f(v)=4\pi \left(\frac{m}{2 \pi k_B T}\right)^{3/2}\cdot \exp \left( -\frac{m v^2}{2k_B T} \right) \cdot  v^2


(a) How big is the most probable velocity $v_m$?  
Die Geschwindigkeit mit der größten Wahrscheinlichkeit ist das Maximum der Maxwellverteilung und folgt somit aus der Gleichung $df(v)/dv=0$. Bevor diese Rechnung durchgeführt wird, führen wir die Variable $a=\frac{m}{2 k_B T}$ ein, damit ergibt sich:
$$
\begin{aligned}
0&=\frac{d}{dv}\left( \frac{4}{\sqrt{\pi}} \cdot a^{3/2} \cdot \exp(-av^2) \cdot v^2 \right) \\
&= \frac{4}{\sqrt{\pi}} \cdot a^{3/2} \frac{d}{dv}\left(\exp(-av^2) \cdot v^2 \right) \\
&= \frac{4}{\sqrt{\pi}} \cdot a^{3/2} \left(   -2av\exp(-av^2)v^2 + 2v\exp(-av^2)  \right) \\
&= \frac{4}{\sqrt{\pi}} \cdot a^{3/2} \cdot \exp(-av^2) (1-av^2) \cdot 2v
\end{aligned}
$$
Also ist das Maximum erreicht für $v_m=\sqrt{1/a}=\sqrt{\frac{2 k_B T}{m}}$.

Die Funktion kann nun umgeschrieben werden zu:
$$
f(v)=\frac{4}{\sqrt{\pi}} \cdot \frac{v^2}{v_m^3} \exp\left(-\frac{v^2}{v_m^2}\right)
$$

(b) How big is the mean of the velocity $\braket{v}$?  
$$
\begin{aligned}
\braket{v}=\int_0^\infty f(v)vdv &= \frac{4}{\sqrt{\pi}} v_m^{-3} \int_0^\infty \exp(-\frac{v^2}{v_m^2}) v^3 dv \\
&=\frac{4}{\sqrt{\pi}} v_m^{-3} \cdot \frac{1}{2} \Gamma(2) \cdot (v_m^2)^2 \\
&=\frac{2}{\sqrt{\pi}}v_m
\end{aligned}
$$

(c) How big is the median of the velocity $v_{0.5}$?  
Gesucht ist 
$$
\begin{aligned}
F(v_{0.5})&=\int_{-\infty}^{v_{0.5}}f(v)dv=0.5 \\
0&=\text{erf}(\beta)-\frac{2}{\sqrt{\pi}}\cdot \beta \cdot  \exp{(-\beta^2)}-0.5
\end{aligned}
$$
mit $\beta=v_{0.5}/v_m$.

In [7]:
import numpy as np
import matplotlib.pyplot as plt
import scipy as sc
from scipy import special
from scipy import optimize 
def f(x):
    return special.erf(x) -2/np.sqrt(np.pi)*x*np.exp(-x**2)- 1/2

In [8]:
median=sc.optimize.root_scalar(f, x0=0, x1=1).root
print(f'Der Median der Geschwindigkeitsverteilung ist {median:.3f} mal v_m')

Der Median der Geschwindigkeitsverteilung ist 1.088 mal v_m


Somit ist der Median der Geschwindigkeit $v_{0.5}\approx 1.088 \cdot v_m$.

(d) How big is the full width at half height of the distribution $(v_\text{FWHM})$?  
Die halbe Höhe entspricht $f(v)=0.5\cdot f(v_m)$. 
$$
0.5 \cdot f(v_m)= \frac{2}{\sqrt{\pi}} v_m^{-3} \exp(-\frac{v_m^2}{v_m^2}) v_m^2=\frac{2}{\sqrt{\pi}v_m} \exp(-1)
$$
Daraus folgt: 
$$
\begin{aligned}
0&= \frac{4}{\sqrt{\pi}} v_m^{-3} \exp(-\frac{v_w^2}{v_m^2}) v_w^2 - \frac{2}{\sqrt{\pi}v_m} \exp(-1) \\
\Rightarrow 0&=2 \gamma^2 \exp(-\gamma^2)-\exp(-1)
\end{aligned}
$$
mit $\gamma=v_w/v_m$.

In [9]:
def g(x):
    return (2*x**2*np.exp(-(x)**2)-np.exp(-1))

v_w1=sc.optimize.root_scalar(g, x0=0, x1=1.1).root
v_w2=sc.optimize.root_scalar(g, x0=1, x1=2).root
print(f'Die linke Punkt ist {v_w1:.3f} mal v_m')
print(f'Der rechte Punkt ist {v_w2:.3f} mal v_m')
print(f'Daraus ergibt sich eine breite von {(v_w2-v_w1):.3f} mal v_m')

Die linke Punkt ist 0.482 mal v_m
Der rechte Punkt ist 1.637 mal v_m
Daraus ergibt sich eine breite von 1.155 mal v_m


(e) How big is the standard deviation of the velocity $\sigma_v$?  
Die Standardabweichung ist die Wurzel aus der Varianz, welcehe sich durch
$$
\text{Var}(V)=E[V^2]-E[V]^2=\sigma^2(V)
$$
berechnen lässt. Der Erwartungswert ist $E[V]=\braket{v}=\frac{2}{\sqrt{\pi}} v_m$. Somit muss noch $E[V^2]$ berechnet werden.
$$
\begin{aligned}
E[V^2]&=\frac{4}{\sqrt{\pi}v_m^3} \int_0^\infty v^4 \exp(-\frac{v^2}{v_m^2})dv \\
&=\frac{4}{\sqrt{\pi}v_m^3} \cdot \frac{1}{2} \frac{\Gamma(2+1/2)}{(v_m^{-2})^{2+1/2}} \\
&=\frac{4}{\sqrt{\pi}v_m^3} \cdot \frac{1}{2} \cdot \frac{3}{4} \sqrt{\pi} v_m^{5}\\
&=\frac{3}{2} v_m^2 
\end{aligned}
$$
Somit ergibt sich für die Standardabweichung:
$$
\begin{aligned}
\Rightarrow \sigma^2&=\frac{3}{2} v_m^2 -v_m^2 =\frac{1}{2}v_m^2 \\
\Rightarrow \sigma&=\frac{1}{4}v_m
\end{aligned}
$$